File: apps/otel-python/otelpython/api/health.py
# SPDX-FileCopyrightText: Copyright (C) Nicolas Lamirault <nicolas.lamirault@gmail.com>
# SPDX-License-Identifier: Apache-2.0

import logging

import fastapi

from otelpython.telemetry import metrics

logger = logging.getLogger(__name__)
router = fastapi.APIRouter()


@router.get("/health", tags=["health"])
async def health_handler() -> dict[str, str]:
    logger.info("[handler] Version")
    metrics.request_counter.add(1, {"target_service": "health"})
    # request_counter.add(1, {"http.route": request.path})
    return {"status": "OK"}

File: apps/otel-js/src/routes/health.js
// SPDX-FileCopyrightText: Copyright (C) Nicolas Lamirault <nicolas.lamirault@gmail.com>
// SPDX-License-Identifier: Apache-2.0

const { getLogger } = require("../telemetry/shared");

async function healthRoutes(fastify) {
  fastify.get("/health", async (request, reply) => {
    const logger = getLogger();
    logger.emit({
      severityText: "info",
      body: "Health handler",
    });
    return reply.send({ status: "ok" });
  });
}

module.exports = healthRoutes;
